src/lib/pickles/pickles.ml

```diff
@@ -3313,6 +3320,20 @@ module Make_str (_ : Wire_types.Concrete) = struct
                     { stmt.proof_state.deferred_values with b = shift_value b }
                 }
             }
+
+          let check_verifier_error err =
+            (* Convert to JSON to make it easy to parse. *)
+            err |> Error_json.error_to_yojson
```

Comment:
wouldn't it make more sense to use a variant to hold all types of errors? (although maybe that's too much work for this PR, but I mean in general)